### ISSUE TITLE
"IndexError: index out of range" internal error for some images with auto rotate enabled

### ISSUE THREAD

**Issue by xtagon** [agschwender/pilbox#28]
Hi,

We've been noticing that certain images are triggering a 500 Internal Server Error when we use the `rotate` op with `deg=auto`. Here's the trace:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/tornado/web.py", line 1334, in _execute
    result = yield result
  File "/usr/lib64/python2.7/site-packages/tornado/gen.py", line 628, in run
    value = future.result()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 109, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib64/python2.7/site-packages/tornado/gen.py", line 633, in run
    yielded = self.gen.send(value)
  File "/usr/lib64/python2.7/site-packages/pilbox/app.py", line 144, in get
    self.render_image(resp)
  File "/usr/lib64/python2.7/site-packages/pilbox/app.py", line 193, in render_image
    outfile, outfile_format = self._process_response(resp)
  File "/usr/lib64/python2.7/site-packages/pilbox/app.py", line 220, in _process_response
    self._image_rotate(image)
  File "/usr/lib64/python2.7/site-packages/pilbox/app.py", line 235, in _image_rotate
    image.rotate(self.get_argument("deg"), **opts)
  File "/usr/lib64/python2.7/site-packages/pilbox/image.py", line 201, in rotate
    exif = self.img._getexif() or dict
  File "/usr/lib64/python2.7/site-packages/PIL/JpegImagePlugin.py", line 390, in _getexif
    return _getexif(self)
  File "/usr/lib64/python2.7/site-packages/PIL/JpegImagePlugin.py", line 429, in _getexif
    info.load(file)
  File "/usr/lib64/python2.7/site-packages/PIL/TiffImagePlugin.py", line 424, in load
    for i in range(i16(fp.read(2))):
  File "/usr/lib64/python2.7/site-packages/PIL/_binary.py", line 37, in i16le
    return i8(c[o]) | (i8(c[o+1]) << 8)
IndexError: string index out of range
```

Minimum Pilbox params to reproduce: `op=rotate&deg=auto` - If I set `deg=90` or some arbitrary degree, the problem doesn't happen.

I suspect there is some edge case with the EXIF data in the images that are failing. If you'd like to inspect one of the failing images I can share it by e-mail. We need it to silently fail if the image doesn't contain orientation metadata.

Regards,
Justin

**Comment by agschwender** [comment]
Looks like it's slated for a future fix in Pillow: https://github.com/python-pillow/Pillow/issues/518. In the meantime, I'll add some exception handling around it.

**Comment by xtagon** [comment]
Thanks @agschwender - much appreciated :+1:

**Comment by agschwender** [comment]
Fixed in 5d97724b64cd79db81e3e5e6a7801cf3cd4dc65c.
